ts/Data/Converters/JSONConverter.ts

```diff
@@ -0,0 +1,260 @@
+import U from '../../Core/Utilities.js';
+const { merge } = U;
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * Handles parsing and transforming JSON to a table.
+ *
+ * @private
+ */
+class JSONConverter extends DataConverter {
+    /* *
+     *
+     *  Static Properties
+     *
+     * */
+
+    /**
+     * Default options
+     */
+    protected static readonly defaultOptions: JSONConverter.Options = {
+        ...DataConverter.defaultOptions,
+        data: [],
+        orientation: 'columns'
+    };
+
+    /* *
+     *
+     *  Constructor
+     *
+     * */
+
+    /**
+     * Constructs an instance of the JSON parser.
+     *
+     * @param {JSONConverter.UserOptions} [options]
+     * Options for the JSON parser.
+     */
+    public constructor(
+        options?: JSONConverter.UserOptions
+    ) {
+        const mergedOptions = merge(JSONConverter.defaultOptions, options);
+
+        super(mergedOptions);
+
+        this.options = mergedOptions;
+        this.table = new DataTable();
+    }
+
+    /* *
+     *
+     *  Properties
+     *
+     * */
+
+    private columns: Array<DataTable.Column> = [];
+    private headers: Array<string> = [];
+    private dataTypes: Array<Array<string>> = [];
+
+    /**
+     * Options for the DataConverter.
+     */
+    public readonly options: JSONConverter.Options;
+
+    private table: DataTable;
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    /**
+     * Initiates parsing of JSON structure.
+     *
+     * @param {JSONConverter.UserOptions}[options]
+     * Options for the parser
+     *
+     * @param {DataEvent.Detail} [eventDetail]
+     * Custom information for pending events.
+     *
+     * @emits JSONConverter#parse
+     * @emits JSONConverter#afterParse
+     */
+    public parse(
+        options: JSONConverter.UserOptions,
+        eventDetail?: DataEvent.Detail
+    ): void {
+
+        const converter = this;
+        options = merge(converter.options, options);
+        const {
+            beforeParse,
+            orientation,
+            firstRowAsNames,
+            columnNames
+        } = options;
+        let data = options.data;
+
+        if (data && beforeParse) {
+            data = beforeParse(data);
+        }
+
+        if (!data) {
+            return;
+        }
+
+        data = data.slice();
+
+        if (orientation === 'columns') {
+            for (let i = 0, iEnd = data.length; i < iEnd; i++) {
+                const item = data[i];
+                if (!(item instanceof Array)) {
+                    return;
+                }
+                if (firstRowAsNames) {
+                    converter.headers.push(`${item.shift()}`);
+                } else if (columnNames) {
+                    converter.headers.push(columnNames[i]);
+                }
+
+                converter.table.setColumn(
+                    converter.headers[i] || i.toString(),
+                    item
+                );
+            }
+        } else {
```

Comment:
In the `else` we should use `else if()` and set `rows` there, otherwise, it will be broken (for instance, someone set orientation:  'cols').